### ISSUE TITLE
Add more filters to members#index

### ISSUE THREAD

**Issue by dan-brewster** [csasf/members#21]
Right now members#index is very basic and mostly useless. Re-write to allow for sorting and filtering multiple categories, and add things like set status, waiver date, and so on. Add a search box with type-ahead filtering by name. Add pagination.

Probably makes sense to use a SPA framework here, at least that's what I would do.

**Comment by wenley** [comment]
I'd say to add this functionality client-side in Javascript, rather than doing these filters/sorting mechanics on the server. Even something hacky that extracts the data from the rendered server response, then re-orders/filters it based on clicks/checkboxes would keep different concerns/behaviors separate.

**Comment by dan-brewster** [comment]
Yep, I agree. I just didn't want to start down that project earlier, but it's worth doing now.

**Comment by dan-brewster** [comment]
This page is actually not used very often, so this is not a high priority at this time.
